NPE when N4JSC calculates dependnecies

```
09:21:01 [build] Exception in thread "main" java.lang.reflect.InvocationTargetException
09:21:01 [build] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
09:21:01 [build] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
09:21:01 [build] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
09:21:01 [build] 	at java.lang.reflect.Method.invoke(Method.java:498)
09:21:01 [build] 	at org.eclipse.jdt.internal.jarinjarloader.JarRsrcLoader.main(JarRsrcLoader.java:61)
09:21:01 [build] Caused by: java.lang.reflect.InvocationTargetException
09:21:01 [build] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
09:21:01 [build] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
09:21:01 [build] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
09:21:01 [build] 	at java.lang.reflect.Method.invoke(Method.java:498)
09:21:01 [build] 	at org.eclipse.n4js.hlc.ProductLauncher.invoke(ProductLauncher.java:196)
09:21:01 [build] 	at org.eclipse.n4js.hlc.ProductLauncher.main(ProductLauncher.java:97)
09:21:01 [build] 	... 5 more
09:21:01 [build] Caused by: java.lang.NullPointerException
09:21:01 [build] 	at org.eclipse.n4js.hlc.base.DependenciesHelper.convertToFilesAddTargetPlatformAndCheckWritableDir(DependenciesHelper.java:158)
09:21:01 [build] 	at org.eclipse.n4js.hlc.base.DependenciesHelper.discoverMissingDependencies(DependenciesHelper.java:54)
09:21:01 [build] 	at org.eclipse.n4js.hlc.base.N4jscBase.doMain(N4jscBase.java:507)
09:21:01 [build] 	at org.eclipse.n4js.hlc.base.N4jscBase.main(N4jscBase.java:325)
09:21:01 [build] 	... 11 more
```

This seems to happen when user invokes N4JSC without project locations.

--- a/tests/org.eclipse.n4js.hlc.tests/src/org/eclipse/n4js/hlc/tests/InstallFromManifestCompileRunN4jscExternalImportsTest.java
+++ b/tests/org.eclipse.n4js.hlc.tests/src/org/eclipse/n4js/hlc/tests/InstallFromManifestCompileRunN4jscExternalImportsTest.java
@@ -74,4 +74,34 @@ public class InstallFromManifestCompileRunN4jscExternalImportsTest extends BaseN
 				out);
 	}
 
+	/**
+	 * Similar to the {@link #testCompileAndRunWithExternalDependencies()} but instead of using
+	 * {@link BuildType#allprojects} with common root, it is using {@link BuildType#projects} with concrete list of
+	 * projects.
+	 */
+	@Test
+	public void testCompileAndRunWithExternalDependencies2() throws IOException, ExitCodeException {
+		final String wsRoot = workspace.getAbsolutePath().toString();
+		final String fileToRun = wsRoot + "/P3/src/f3.n4jsx";
+
+		final String[] args = {
+				"--systemLoader", COMMON_JS.getId(),
+				"--installMissingDependencies",
+				"-rw", "nodejs",
+				"-r", fileToRun,
+				// "--verbose",
+				// "--debug",
+				"-bt", BuildType.projects.toString(), wsRoot + "/P1", wsRoot + "/P2", wsRoot + "/P3"
+		};
+		final String out = runAndCaptureOutput(args);
+		N4CliHelper.assertExpectedOutput(
+				"P1\n" +
+						"react is not undefined true\n" +
+						"react-dom is not undefined true\n" +
+						"imports from libs are different true\n" +
+						"P2\n" +
+						"React is not undefined true",
+				out);
+	}
+
 }

--- a/tools/org.eclipse.n4js.hlc.base/src/org/eclipse/n4js/hlc/base/DependenciesHelper.java
+++ b/tools/org.eclipse.n4js.hlc.base/src/org/eclipse/n4js/hlc/base/DependenciesHelper.java
@@ -50,11 +50,13 @@ class DependenciesHelper {
 	Map<String, String> discoverMissingDependencies(String projectLocations, List<File> srcFiles) {
 
 		List<File> allProjectsRoots = new ArrayList<>();
+		List<File> discoveredProjectLocations = new ArrayList<>();
 
-		List<File> locations = convertToFilesAddTargetPlatformAndCheckWritableDir(projectLocations);
-		// Discover projects in search paths.
-		List<File> discoveredProjectLocations = collectAllProjectPaths(locations);
-
+		if (projectLocations != null) {
+			List<File> locations = convertToFilesAddTargetPlatformAndCheckWritableDir(projectLocations);
+			// Discover projects in search paths.
+			discoveredProjectLocations = collectAllProjectPaths(locations);
+		}
 		// Discover projects for single source files.
 		List<File> singleSourceProjectLocations = new ArrayList<>();
 		try {

